packages/bot-web-ui/src/components/dashboard/tour-trigger-dialog.tsx

```diff
@@ -9,58 +9,107 @@ type TourTriggrerDialog = {
     is_tour_dialog_visible: boolean;
     setTourDialogVisibility: (param: boolean) => void;
     setOnBoardTourRunState: (param: boolean) => void;
+    setBotBuilderTourState: (param: boolean) => void;
+    active_tab: number;
 };
 
 const TourTriggrerDialog = ({
     is_tour_dialog_visible,
     setTourDialogVisibility,
+    setBotBuilderTourState,
     setOnBoardTourRunState,
     setTourActive,
+    active_tab,
 }: TourTriggrerDialog) => {
     const handleChange = () => {
-        setTourActive(true);
-        setOnBoardTourRunState(true);
+        if (active_tab === 0) {
+            setTourActive(true);
+            setOnBoardTourRunState(true);
+        } else {
+            setBotBuilderTourState(true);
+        }
         setTourDialogVisibility(false);
     };
-    const closTourChange = () => {
+    const closeTourChange = () => {
+        if (active_tab === 0) {
+            setTourActive(false);
+            setOnBoardTourRunState(false);
+        } else {
+            setBotBuilderTourState(false);
+        }
+
```

Comment:
@farabi-deriv Could you pls write a reusable function for it to avoid duplication? 
for example:
```
const [anyFunction] = (**value**: boolean) => {
    if (active_tab === 0) {
                setTourActive(**value**);
                setOnBoardTourRunState(**value**);
            } else {
                setBotBuilderTourState(**value**);
            }
}
```

Reply:
Alright. Done.